shell/platform/android/external_view_embedder/external_view_embedder_unittests.cc

```diff
@@ -2,17 +2,53 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include "flutter/flow/surface.h"
 #include "flutter/fml/raster_thread_merger.h"
 #include "flutter/fml/thread.h"
 #include "flutter/shell/platform/android/external_view_embedder/external_view_embedder.h"
+#include "flutter/shell/platform/android/jni/jni_mock.h"
+#include "flutter/shell/platform/android/surface/android_surface_mock.h"
+#include "gmock/gmock.h"
 #include "gtest/gtest.h"
+#include "third_party/skia/include/gpu/GrContext.h"
 
 namespace flutter {
 namespace testing {
 
+using ::testing::ByMove;
+using ::testing::Return;
+
+class SurfaceMock : public Surface {
```

Comment:
This can be moved to flow if the general pattern looks ok, so other unit tests can use the mock.